bitcoin/src/util/hash.rs

```diff
@@ -39,6 +39,44 @@ where
 
 /// Calculates the merkle root of an iterator of *hashes*.
 ///
+/// This fcuntion calculate the root from a merkle path when we need to derive the root from the
```

Comment:
typo: fcuntion, here and some lines below

Reply:
Also "calculates" is missing "s" at the end. There should be a comma or dot and a new sentence before "when". Also the commit message has a typo - "merle"